### PR TITLE
Create Stas

### PR DESCRIPTION
**Pull request by stas83ggg**
I don't know how to add files to path-1

### CODE REVIEW

File: Ball.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+public class Ball {
+    String color;
+    int number;
```
Comment (uliana-andrushchuk) [inline Ball.java:3]: Please, add access modifiers

File: ColorSupplier.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import java.util.Random;
+
+public class ColorSupplier {
+    public Colors getRandomColor(){
+       int index = new Random().nextInt(Colors.values().length);
```
Comment (uliana-andrushchuk) [inline ColorSupplier.java:5]: Please, make `new Random()` class field

File: Colors.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+public enum Colors {
```
Comment (uliana-andrushchuk) [inline Colors.java:1]:
```suggestion
public enum Color {
```
Please, use singular form for enum name

File: Lottery.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import java.util.Random;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        Ball ball = new Ball();
+        int index = new Random().nextInt(Colors.values().length);
+        int value = new Random().nextInt(100);
```
Comment (uliana-andrushchuk) [inline Lottery.java:7]: Please, make `new Random()` class field

Comment (uliana-andrushchuk) [reply Lottery.java:7]: Make `100` constant

File: Main.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+public class Main {
+    public static void main(String[] args) {
+       for (int i = 0; i < 3; i++) {
+           Lottery lottery = new Lottery();
```
Comment (uliana-andrushchuk) [inline Main.java:4]: Please, do not create new Lottery() inside the loop

**Review by uliana-andrushchuk — CHANGES_REQUESTED** [review]

File: ColorSupplier.java (hunk 2 above)
Comment (stas83ggg) [reply ColorSupplier.java:5]: Could You write some example, please ?

File: Ball.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
+public class Ball {
+    public String color;
+    public int number;
```
Comment (uliana-andrushchuk) [inline Ball.java:3]: Class fields should be `private`

File: ColorSupplier.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import java.util.Random;
+
+public class ColorSupplier {
+    public Color getRandomColor(int index) {
```
Comment (uliana-andrushchuk) [inline ColorSupplier.java:4]:
```suggestion
    public Color getRandomColor() {
```
You do not need to pass index to this method

File: Stas [hunk 8]
```diff
@@ -0,0 +1 @@
+I don't know how to add files to path-1
```
Comment (uliana-andrushchuk) [inline Stas:1]: What do you mean?
Please, remove this file

File: Ball.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
```
Comment (uliana-andrushchuk) [inline Ball.java:4]: Please, create getters and setters or constructor with parameters

File: ColorSupplier.java (hunk 2 above)
Comment (uliana-andrushchuk) [reply ColorSupplier.java]: `private static Random random = new Random();`
add this field to ColorSupplier class

**Review by uliana-andrushchuk — CHANGES_REQUESTED** [review]

**Comment by stas83ggg** [comment]
Uliana, I added new version-3 of files. Could You check ?

**Comment by stas83ggg** [comment]
Снимок экрана 2023-04-26 в 20 41 16

Uliana, I made all changes which You recommended but code does not work (((

File: ColorSupplier.java [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import java.util.Random;
+
+public class ColorSupplier {
+    private static Random random = new Random();
+    public Color getRandomColor() {
+       int index = new Random().nextInt(Color.values().length);
```
Comment (uliana-andrushchuk) [inline ColorSupplier.java:6]:
```suggestion
       int index = random.nextInt(Color.values().length);
```

File: Main.java [hunk 11]
```diff
@@ -0,0 +1,8 @@
+public class Main {
+    public static void main(String[] args) {
+       Lottery lottery = new Lottery();
+        for (int i = 0; i < 3; i++) {
+           System.out.println(lottery);
```
Comment (uliana-andrushchuk) [inline Main.java:5]:
```suggestion
           System.out.println(lottery.getRandomBall());
```

File: Lottery.java [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import java.util.Random;
+
+public class Lottery {
+    private static final int LOTTERY_MAX_COUNT = 100;
+    public Ball getRandomBall(int index, int value) {
```
Comment (uliana-andrushchuk) [inline Lottery.java:5]:
```suggestion
    public Ball getRandomBall() {
```
This method should be without input parameters

File: Lottery.java [hunk 13]
```diff
@@ -0,0 +1,11 @@
...
+        Ball ball = new Ball();
+        index = new Random().nextInt(Color.values().length);
```
Comment (uliana-andrushchuk) [inline Lottery.java:7]:
```suggestion
```

File: Lottery.java [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+        value = new Random().nextInt(LOTTERY_MAX_COUNT);
```
Comment (uliana-andrushchuk) [inline Lottery.java:8]: add ball.setNumber and ball.setColor()

**Review by uliana-andrushchuk — CHANGES_REQUESTED** [review]

File: Lottery.java [hunk 15]
```diff
@@ -0,0 +1,13 @@
+import java.util.Random;
+
+public class Lottery {
+   private static final int LOTTERY_MAX_COUNT = 100;
```
Comment (uliana-andrushchuk) [inline Lottery.java:4]: >     private final ColorSupplier colorSupplier = new ColorSupplier();
>     private final Random random = new Random();

Please, add these fields here, and use them inside getRandomBall() method

File: Lottery.java [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+    public Ball getRandomBall() {
+        Ball ball = new Ball();
+        int number = new Random().nextInt(LOTTERY_MAX_COUNT);
+        ball.setNumber(number);
+        ColorSupplier supplier = new ColorSupplier();
```
Comment (uliana-andrushchuk) [inline Lottery.java:9]:
```suggestion
```

File: ColorSupplier.java [hunk 17]
```diff
@@ -0,0 +1,9 @@
+import java.util.Random;
+
+public class ColorSupplier {
+    private static Random random = new Random();
```
Comment (uliana-andrushchuk) [inline ColorSupplier.java:4]:
```suggestion
    private final Random random = new Random();
```

File: Lottery.java [hunk 18]
```diff
@@ -0,0 +1,13 @@
+import java.util.Random;
+
+public class Lottery {
+   private static final int LOTTERY_MAX_COUNT = 100;
+    public Ball getRandomBall() {
+        Ball ball = new Ball();
+        int number = new Random().nextInt(LOTTERY_MAX_COUNT);
```
Comment (uliana-andrushchuk) [inline Lottery.java:7]:
```suggestion
        int number = random.nextInt(LOTTERY_MAX_COUNT);
```

File: Lottery.java [hunk 19]
```diff
@@ -0,0 +1,13 @@
...
+        ball.setNumber(number);
+        ColorSupplier supplier = new ColorSupplier();
+        ball.setColor(String.valueOf(supplier.getRandomColor()));
```
Comment (uliana-andrushchuk) [inline Lottery.java:10]: Please, do not use `String.valueOf` with ENUM values 
Use name() for getting String representation of enum constants
https://mate-academy.github.io/jv-program-common-mistakes/java-core/oop/lottery#use-name-for-getting-string-representation-of-enum-constants

**Review by uliana-andrushchuk — CHANGES_REQUESTED** [review]

File: Lottery.java (hunk 19 above)
Comment (stas83ggg) [reply Lottery.java:10]: Uliana, thank You very much for improving my Java's skills ! :)

**Review by olekskov — APPROVED** [review]
Good job in general, approved.
I see that all your files lack empty line in the end of file. Please check following link to make your IDE to put it automatically:
https://stackoverflow.com/questions/16761227/how-to-make-intellij-idea-insert-a-new-line-at-every-end-of-file

**Pull request closed without merging**